process/main.py

```diff
@@ -568,6 +567,34 @@ def output(self):
         """write model output"""
 
 
+class CostManager(CostsProtocol):
```

Comment:
I dont think the CostManager should inherit from the protocol
```suggestion
class CostManager:
```